ClientFactory/MockFactory.php

```diff
@@ -10,14 +11,16 @@
 class MockFactory implements ClientFactory
 {
     /**
-     * @var Client
+     * @var HttpClient
      */
     private $client;
 
     /**
-     * @param Client $client
+     * Set the client instance that this factory should return.
+     *
+     * Note that this can be any client, not only a mock client.
      */
-    public function setClient(Client $client)
+    public function setClient(HttpClient $client)
```

Comment:
This is technically a BC break.

But if I understand it correctly. Using this mock factory has never worked, right?
If so, it would mean that this is a bugfix and I'm 👍